tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -639,6 +639,25 @@ public void testDescribe(String quorum) throws ExecutionException, InterruptedEx
         assertTrue(rows[0].startsWith(String.format("Topic: %s", testTopicName)), "Row does not start with " + testTopicName + ". Row is: " + rows[0]);
     }
 
+    @ParameterizedTest(name = ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"quorum=zk", "quorum=kraft"})
+    public void testDescribeWithDescribeTopicPartitionsApi(String quorum) throws ExecutionException, InterruptedException {
+        TestUtils.createTopicWithAdmin(adminClient, testTopicName, scalaBrokers, scalaControllers, 3, (short) 2,
+            scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
+        );
+        String secondTopicName = "test-2";
+        TestUtils.createTopicWithAdmin(adminClient, secondTopicName, scalaBrokers, scalaControllers, 3, (short) 2,
+            scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
+        );
+
+        String output = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap(
+            "--describe", "--partition-size-limit-per-response=1"));
+        String[] rows = output.split("\n");
+        assertEquals(8, rows.length, String.join("\n", rows));
+        assertTrue(rows[2].contains("\tElr"), rows[2]);
+        assertTrue(rows[2].contains("LastKnownElr"), rows[2]);
+    }
+
     @ParameterizedTest(name = ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME)
```

Comment:
I'd like to see an integration test here that utilizes pagination. Something like 3 topics with various number of partitions between 50 and 100. Then use a page size of 20 or so.

Reply:
Thanks for the advice. Updated.